mava/wrappers/jumanji.py

```diff
@@ -117,3 +118,73 @@ def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
 
         # Aggregate the list of individual rewards and use a single team_reward.
         return self.aggregate_rewards(timestep, modified_observation)
+
+
+class ConnectorWrapper(MultiAgentWrapper):
+    """Multi-agent wrapper for the MA Connector environment."""
+
+    def __init__(self, env: MaConnector):
+        super().__init__(env)
+
+    def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
+        """Modify the timestep for the Connector environment."""
+
+        def create_agents_view(grid: chex.Array) -> chex.Array:
+            positions = jnp.where(grid % 3 == 2, True, False)
+            targets = jnp.where((grid % 3 == 0) & (grid != 0), True, False)
+            paths = jnp.where(grid % 3 == 1, True, False)
+            my_position = jnp.where(grid == 2, True, False)
+            my_target = jnp.where(grid == 3, True, False)
+            agents_view = jnp.stack((positions, targets, paths, my_position, my_target), -1)
+            return agents_view
+
+        def create_global_state(grid: chex.Array) -> chex.Array:
+            positions = jnp.where(grid % 3 == 2, True, False)
+            targets = jnp.where((grid % 3 == 0) & (grid[0] != 0), True, False)
+            paths = jnp.where(grid % 3 == 1, True, False)
+            global_state = jnp.stack((positions, targets, paths), -1)
+            return global_state
```

Comment:
Seeing as the gird is already all agents views can we just use `grid[0]` here? And then tile the output so that we have a single global state?